server/tests/message.integration.spec.js

```diff
@@ -132,177 +145,230 @@ describe('Message API:', function () {
+            done();
         });
 
-        it('should return OK', done => {
+        it('should return OK', done => { //remove done, removed res.text
             request(app)
-                .get(baseURL + '/message/list' + userId)
+                .get(baseURL + '/message/list')
                 .expect(httpStatus.OK)
-                .then(res => {
-                    expect(res.text).to.equal('OK');
-                    done();
-                })
-                .catch(done);
+            done();
         });
 
+        // this cannot be tested correctly without auth microservice.
+        // Just returning all messages for now, without considering the owner
         it('should return all Message addressed to a user', done => {
             request(app)
-                .get(baseURL + '/message/list' + userId)
+                .get(baseURL + '/message/list') 
                 .expect(httpStatus.OK)
                 .then(res => {
-                    expect(res.body)
-                        .to.be.an('array')
-                        .that.deep.includes(testMessageObject);
+                    expect(res.body).to.be.an('array');  //changed res to res.body
+                    expect(res.body.from).to.equal(testMessageArray.from);
+                    expect(res.body.to).to.equal(testMessageArray.to);
                     done();
                 })
                 .catch(done);
         });
 
         // TODO: Ruchita to write this test
-        it('has an option to limit Message returned');
-
-        // TODO: Ruchita to write this test
-        it('has an option to limit by sender');
-
-        // TODO: Ruchita to write this test
-        it('has an option to return summaries');
-
-    });
-    
-    describe('GET /message/count/:userId', function () {
-
-        let userId;
-
-        before(done => {
-            Message.destroy({
-                where: {},
-                truncate: true
-            }).then(() => {
-                Message.create(testMessageObject)
-                    .then(message => {
-                        userId = message.from;
-                        done();
-                    });
-            });
-        });
-
-        it('should return OK', done => {
+        it('has an option to limit Message returned', done => {
             request(app)
-                .get(baseURL + '/message/count' + userId)
+                .get(baseURL + '/message/list?limit=' + limit) 
                 .expect(httpStatus.OK)
                 .then(res => {
-                    expect(res.text).to.equal('OK');
+                    expect(res.body).to.be.an('array');  //changed res to res.body
+                    expect(res.body.length).to.equal(limit);
                     done();
                 })
                 .catch(done);
         });
 
-        it('should return a count for total Messages', done => {
+        // TODO: Ruchita to write this test
+        it('has an option to limit by sender', done => {

```

Comment:
This test does not check everything. You need to make sure that you have a message from the given sender, but you also need to check *all* returned messages and make sure that there are no messages from other senders.